server.py

```diff
@@ -20,26 +19,32 @@ def get_ids(self):
         return self.__boards.keys()
 
     def add_board(self, room_id, board):
-        self.__boards[room_id] = board
+        if room_id in self.__boards:
+            assert ValueError('Room with given ID already exists')
+        else:
+            self.__boards[room_id] = board
 
     def get_board(self, room_id):
         return self.__boards[room_id]
 
     def update_board(self, diffs: list, room_id: str):
-        self.__boards[room_id].update_board(diffs)
+        if room_id not in self.__boards:
+            assert ValueError('Room with given ID does not exist')
+        else:
+            self.__boards[room_id].update_board(diffs)
+
 server = Server()
 
 @app.route('/')
 def index():
     return app.send_static_file('index.html')
 
-@app.route('/create')
-def create_room(payload):
-    room = payload['room_id']
-    if room in server.get_ids():
-        abort(Response('Room with given ID already exists'))
-    server.add_board(room, CanvasBoard.create_board(WIDTH, HEIGHT))
-    response = jsonify(room_id=room)
+@app.route('/create/<room_id>')
+def create_room(room_id):
+    if room_id in server.get_ids():
+        abort(Response('Room with given ID already exists', status=400))
+    server.add_board(room_id, CanvasBoard.create_board(WIDTH, HEIGHT))

```

Comment:
We could simply try and catch here without checking it explicitly since the underlying method checks for this already.
Then we can get rid of the the get_ids method.

Same thing for send-stroke.

Reply:
Revised